18764-Remove-external-dependencies-from-Kernel-Tests-

demarey opened the pull request:
https://pharo.fogbugz.com/f/cases/18764/Remove-external-dependencies-from-Kernel-Tests

demarey committed:
https://pharo.fogbugz.com/f/cases/18764/Remove-external-dependencies-from-Kernel-Tests

demarey commented:
redo stuff done in #365 with current image.
@guillep It looks like Directory class you introduced is not needed anymore (no more dependency on FileSystem in testPrimitiveNamed2. Could you confirm this?
```smalltalk
testPrimitiveNamed2
	"This test useses the #primPathNameDelimiter primitive."

	self compile: '<primitive: ''primitiveDirectoryDelimitor'' module: ''FilePlugin''> ^ #delim' selector: #delim.
	self assert: self delim equals: File delimiter.
```
I did not include Directory class and its test class in this pull request but could be done if needed.

Ducasse commented:
The error does not seem related. I do not know how to address the error and I think that we should put our heads together and fix it.

pavel-krivanek commented on src/DelaySchedulerDebugUtilities/package.st:
```diff
@@ -0,0 +1 @@
+Package { #name : #DelaySchedulerDebugUtilities }
```
The new package that is not mentioned in baselines...

Ducasse replied:
Christophe can you check?

demarey committed:
Add DelaySchedulerDebugUtilities to BaselineOfIDE

demarey committed:
merge with pharo-project/development

demarey committed:
apply fix on the merge

demarey commented:
@pavel-krivanek I added the package to BaselineOfIDE and did a merge to avoid conflicts with current state of pharo development branch

pavel-krivanek commented:
The CI reports some conflicts

Commit:
Merge branch 'Pharo7.0' into demarey/18764-Remove-external-dependencies-from-Kernel-Tests-

pavel-krivanek commented:
related failure: DependenciesTests.SystemDependenciesTest

pavel-krivanek commented:
SystemDependenciesTest>>#testExternalIDEDependencies

pavel-krivanek commented on src/BaselineOfIDE/BaselineOfIDE.class.st:
```diff
@@ -178,6 +178,7 @@ BaselineOfIDE >> baseline: spec [
 		spec package: 'Ring-Deprecated-Tests-Containers'.
 		spec package: 'Ring-Deprecated-Tests-Kernel'.
 		spec package: 'System-DependenciesTests'.
+		spec package: 'DelaySchedulerDebugUtilities'.
```
The package is not mentioned in the baseline groups

tesonep committed:
Update BaselineOfIDE.class.st

Putting the DelaySchedulerDebugUtilities package in a group so it will be loaded

guillep commented:
I'd move it to Pharo8

Ducasse commented:
I relaunched the build

Ducasse commented:
Failing tests are unrelated!

Ducasse commented:
Code looks simple! Thanks for your patience!

estebanlm commented:
this was incorrectly merged into Pharo 7.0. I will revert and please can we re-submit this for Pharo 8?
